src/Okta.Sdk/Internal/ResourceTypeResolverFactory.cs

```diff
@@ -31,16 +31,24 @@ internal static class ResourceTypeResolver
             .Select(typeInfo => (typeInfo.AsType(), typeInfo.BaseType.GenericTypeArguments[0]))
             .ToArray();
 
+        public static Type GetResolver(Type resourceType)
+            => CachedResolvers.FirstOrDefault(x => x.For == resourceType).Resolver;
+
         public static AbstractResourceTypeResolver<T> Create<T>()
         {
-            var resolver = CachedResolvers.FirstOrDefault(x => x.For == typeof(T)).Resolver;
+            var resolverType = GetResolver(typeof(T));
 
-            if (resolver == null)
+            if (resolverType == null)
             {
                 return new DefaultResourceTypeResolver<T>();
             }
 
-            return (AbstractResourceTypeResolver<T>)Activator.CreateInstance(resolver);
+            return (AbstractResourceTypeResolver<T>)Activator.CreateInstance(resolverType);
+        }
+
+        public static IResourceTypeResolver Create(Type resourceType)
+        {
+            return (IResourceTypeResolver)Activator.CreateInstance(resourceType);
```

Comment:
Is this doing the same thing as line 46?

Reply:
Not exactly the same, here we cast to the new interface `IResourceTypeResolver`. The one from line 46 is returning objects cast as the abstract class.

I kept both, but we can opt to use only that returns `IResourceTypeResolver`